src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -43,11 +49,61 @@ define([
         deploymentsFetchState: States.STATE_LOADING,
         fetchState: States.STATE_LOADING,
         modalClass: null,
+        route: null,
         tasksFetchState: States.STATE_LOADING
       };
     },
 
     componentDidMount: function() {
+      var router = this.props.router;
+
+      function callRoute(routeFunc) {
+        if(this.state.fetchState === States.STATE_LOADING) {
+          setTimeout(callRoute.bind(this, routeFunc), 300);
```

Comment:
If you come in by http://localhost:8080/#app/sleep30/configuration directly the model isn't loaded yet, that takes some time, up to seconds.
This functions checks every 300 mseconds if the model is still loading and if not loading anymore then sets the state for the correct route.